swo_aws_extension/flows/jobs/billing_journal/processor_dispatcher.py

```diff
@@ -0,0 +1,79 @@
+from swo_aws_extension.flows.jobs.billing_journal.line_processors.marketplace import (
+    GenerateMarketplaceJournalLines,
+)
+from swo_aws_extension.flows.jobs.billing_journal.line_processors.other_services import (
+    GenerateOtherServicesJournalLines,
+)
+from swo_aws_extension.flows.jobs.billing_journal.line_processors.saving_plans import (
+    GenerateSavingPlansIncentivateJournalLines,
+    GenerateSavingPlansJournalLines,
+)
+from swo_aws_extension.flows.jobs.billing_journal.line_processors.support import (
+    GenerateSupportDevelopmentJournalLines,
+    GenerateSupportEnterpriseJournalLines,
+    GenerateSupportJournalLines,
+)
+from swo_aws_extension.flows.jobs.billing_journal.line_processors.usage import (
+    GenerateUpfrontIncentivateJournalLines,
+    GenerateUpfrontJournalLines,
+    GenerateUsageIncentivateJournalLines,
+    GenerateUsageJournalLines,
+)
+
+
+class JournalProcessorDispatcher:
+    """Dispatcher for journal line processors based on itemSku."""
+
+    def __init__(self, processors):
+        self.processors = list(processors)
+
+    def process(self, item_sku, account_id, account_metrics, journal_details, account_invoices):
+        """Process the item SKU using the appropriate processors.
+
+        Args:
+            item_sku (str): The item SKU to process.
+            account_id (str): The account ID.
+            account_metrics (dict): Metrics related to the account.
+            journal_details (dict): Details for the journal entry.
+            account_invoices (dict): Invoices related to the account.
+
+        Returns:
+            list: List of journal lines generated by the processors.
+        """
+        journal_lines = []
+        for processor in self.processors:
+            if processor.can_process(item_sku):
+                journal_lines.extend(
+                    processor.process(
+                        account_id, account_metrics, journal_details, account_invoices
+                    )
+                )
+        return journal_lines
+
+
+def create_journal_dispatcher(config):
```

Comment:
We can follow the same approach I mentioned before about the build classmethod. The `JournalProcessosDispatcher` knows how it should build itself, so we can add a `build` method inside the class instead of creating a method outside. This way, we keep the responsability inside the class. WDYT?